lib/util/utils.go

```diff
@@ -187,6 +187,7 @@ func AsService(fn func(stop chan struct{})) suture.Service {
 type ServiceWithError interface {
 	suture.Service
 	Error() error
+	SetError(error)
 }
```

Comment:
Starting to realize util.Service should have passed in a context oppose to a channel.

Reply:
That would definitely be useful in places, but not happening in this PR (just due to time).

Reply:
It's a fairly straghtforward refactor to do separately.